### PR TITLE
Data Reference sidebar improvements

### PR DESCRIPTION
**Pull request by mazameli**
This PR makes the data reference side panel more useful for when you're writing SQL. It assumes that the data reference is not very useful or hardly used when creating custom GUI queries. 

**Current changes include:**
- Displaying the actual underlying names of tables and columns
- Removed "potentially interesting questions" and the lists of saved aggregations ("metrics") and segments from table detail panels. These aren't helpful when writing SQL since they take you to the GUI builder, and in the case of metrics and segments, can't be referenced in SQL queries. For GUI questions, we've also come up with better solutions for these items: drill through, filtering by a segment, summarizing by a metric, etc. (In the future maybe we could consider adding potentially useful SQL snippets here.)
- When you open the data reference, if you have a database selected for a SQL question, or a table selected for a GUI question, the data reference will initially display that DB/table. (#1877 / #8307)
- A first stab at displaying sample values for columns. Currently only works for columns that are set to `List`.

**Cosmetic improvements:**
- Tightened up typography
- Added hover states to click targets
- Smoothed out transition animation when the panel slides out
- Improved the look of the count of tables for each DB
- Replaced our `table2` icon with a simpler, nicer looking one that looks less like a calculator.
- Replaced our `field` icon with one that looks less like a dissolving pencil.

**Changes I'd like us to make in other PRs:**
- Add support for listing schemas (related to https://github.com/metabase/metabase/issues/7926 but not exactly the same since this is the full-page data reference)
- Improve the sample values behavior:
-- Make it work for all column types except for `Descriptions` e.g. blog posts, comments, etc.
-- Only list 10 distinct values.
- Make the `Back` button's microcopy dynamic: display the name of the parent item, or display "all databases" if currently filtering to the selected database.
- The way we currently display "connections" for a table feels backwards to me. When I'm viewing the Orders table, I see 0 connections, but I'd expect to see something like `user_id --> people` and `product_id --> products` — i.e. a list of what the foreign keys are. I think right now it's showing the reverse, e.g. when looking at the Products table, I see the tables that have FKs pointing to _it_.
- Prevent "Saved Questions" from being shown as a database (https://github.com/metabase/metabase/issues/9458)

**Screenshots**

screen shot 2019-02-20 at 1 51 32 pm

screen shot 2019-02-20 at 1 51 51 pm

screen shot 2019-02-20 at 1 52 06 pm

### CODE REVIEW

**Comment by mazameli** [comment]
I'm seeing this now when I click into a field. Should I expect to see the field's name and description along with that `Card` with the values in it, or do I need to reimplement that? A cursory glance at the diff in the last couple commits leads to me to believe I should be seeing something more than this:

image

**Comment by tlrobinson** [comment]
@mazameli My commit only changed which panel is initially displayed (I had @kdoh revert the 2nd commit with my prototype of the field values preview). @kdoh's commit just filled in your previous commit with the real field values but otherwise it's how you left it.

File: frontend/src/metabase/query_builder/components/dataref/DatabasePane.jsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+import MainPane from "./MainPane";
+
+const DatabasePane = ({ database, ...props }) => (
+  <MainPane {...props} databases={[database]} />
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/dataref/DatabasePane.jsx]: Did you decide not to do a different pane for a single database vs. all the databases?

**Review by tlrobinson — APPROVED** [review]

File: frontend/src/metabase/query_builder/components/dataref/DatabasePane.jsx [hunk 2]
```diff
@@ -1,9 +1,47 @@
...
+        <Icon name="table2" className="text-light pr1" size={12} />
+        <span className="text-medium">{database.tables.length}</span>
+      </div>
+    </div>
+
+    <ul>
+      {database.tables.filter(isQueryable).map((table, index) => (
+        <li
+          key={table.id}
+          className={cx("", {
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/dataref/DatabasePane.jsx]: You can just remove this whole thing (`className` and the following 2 lines)

File: frontend/src/metabase/query_builder/components/dataref/DetailPane.jsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ const DetailPane = ({ name, description, extra, values }) => (
       <Icon name="field" className="text-medium pr1" size={16} />
       <h3>{name}</h3>
     </div>
-    <p className={cx({ "text-medium": !description })}>
+    <p className={"text-spaced " + cx({ "text-medium": !description })}>
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/dataref/DetailPane.jsx]: `cx` combines all of it's arguments so this is slightly preferable:

```
<p className={cx("text-spaced", { "text-medium": !description })}>
```

File: frontend/src/metabase/query_builder/components/dataref/DatabasePane.jsx [hunk 4]
```diff
@@ -1,9 +1,47 @@
...
+            {table.name}
+          </a>
+        </li>
+      ))}
+    </ul>
+  </div>
 );
 
+DatabasePane.propTypes = {
+  show: PropTypes.func.isRequired,
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/dataref/DatabasePane.jsx]: technically should also have `database: PropTypes.object.isRequired,`

**Comment by mazameli** [comment]
@salsakran This should be ready to bear the weight of your withering gaze if you want to give it a spin.

**Review by salsakran — APPROVED** [review]
Looks great!

**Pull request merged**